### ISSUE TITLE
Allow multiple contents to be put via API

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#20060]
dotCMS has REST apis for saving content in the system, most notably, the content and workflowAPIs.  This allows for single pieces of content to be put one at a time in dotCMS.

In addition to these endpoints, we need another endpoint that can take multiple contents/assets.  The format for these puts should be similar to our current workflow API put, where you can specify a workflow and a contentlet as a json object - but instead it should take a workflow and then array of content objects and loop over them (perhaps multithreaded?) and ingest them in dotCMS.   Each content object created would get pushed through the same workflow step/process.

If we mutlithreaded, we should 1. make that configurable and 2. start with say 4 threads.

**Comment by fmontes** [comment]
Right now we do this to create ONE piece of content:

```
fetch('/api/v1/workflow/actions/default/fire/PUBLISH', {
    method: 'PUT',
    body: JSON.stringify({ "contentlet": { "title": "Re: 10 Surfing spots to checkout in Tahiti", "contentHost": "demo.dotcms.com", "contentType": "BlogComment", "blog": "e7ceee89-3cc5-4e35-9795-5927ac88ad89", "postDate": "02/26/21 19:13:14", "commentAuthor": "86fe5be1-4624-4595-bf2d-af8d559414b1", "body": "test" } }),
    headers: {
        'Content-type': 'application/json; charset=UTF-8'
    }
})
    .then(res => res.json())
    .then(console.log)
```

We can do the same thing to create several but instead of taking an object, it should receive an array of objects.

Also, we should take the property `contentlet` and `contentlets` any of those should take an array OR just an object.

### For binary fields
For files we are going to get URLs, we can use the `temp` API to create those and assign them to the contentlets.

Maybe we can check the extension of the URL to assert the type of file and if now we might need to hit the URL which will make it slower but is ok.

**Comment by wezell** [comment]
What should the return code be?
200?
202?
207 Multi-Status

**Comment by john-thomas-dotcms** [comment]
@bryanboza we're going to want to make sure that using this to create large numbers of content doesn't cause performance issues.

**Comment by erickgonzalez** [comment]
I was able to create almost 300 contentlets in 9s.
```json
"summary": {
            "affected": 301,
            "failCount": 0,
            "successCount": 301,
            "time": 8933
        }
```

**Comment by bryanboza** [comment]
If you try to send an array with some errors, we are not creating the pieces of content, we are returning 200 as response code and also we are not returning an error message why the content was not created 

We need some work here: 
![image](https://user-images.githubusercontent.com/2641437/130110070-d1eeb8b6-1a84-49fc-8a8c-1a7fbd3eb9ba.png)

**Comment by jdotcms** [comment]
I have made a fix
https://github.com/dotCMS/core/pull/20825/files

The result now is

```
{
    "entity": {
        "results": [
            {
                "{title=title2, contentHost=demo.dotcms.com, body=test2}": {
                    "errorMessage": "Content Type does not exist",
                    "identifier": null,
                    "inode": null
                }
            },
            {
                "{title=title3, contentHost=demo.dotcms.com, body=test3}": {
                    "errorMessage": "Content Type does not exist",
                    "identifier": null,
                    "inode": null
                }
            },
            {
                "{title=title1, contentHost=demo.dotcms.com, body=test1}": {
                    "errorMessage": "Content Type does not exist",
                    "identifier": null,
                    "inode": null
                }
            }
        ],
        "summary": {
            "affected": 3,
            "failCount": 3,
            "successCount": 0,
            "time": 29923
        }
    },
    "errors": [],
    "i18nMessagesMap": {},
    "messages": [],
    "permissions": []
}
```

**Comment by jdotcms** [comment]
this example shows, how to send binary files as part of the multiple fires
```
{
    "contentlets":[
        {
            "contentType":"BinaryTest",
            
            "name":"test1",
            "book":"temp_64bb102d49"    
        }
    ]
}
```
in this clase, book is the binary and "temp_64bb102d49" is the temporal file id

**Comment by bryanboza** [comment]
Fixed, tested on release-21.06 // Postgres // FF
